modules/core/src/lib/layer.js

```diff
@@ -476,15 +478,22 @@ export default class Layer extends Component {
     const cacheSize = pickingColorCache.length / 3;
 
     if (cacheSize < numInstances) {
+      let overAlloc;
+      if (numInstances > MAX_PICKING_COLOR_CACHE_SIZE) {
+        log.warn('Index out of picking color range. Picking might not work as expected.');
```

Comment:
```suggestion
        log.warn('Layer has too many rows. Picking might not be able to distinguish all rows.');
```

Reply:
- We do not use the term "rows" in documentations. At least say "data has too many rows" or "layer has too many data objects."
- `log.warn` returns a function. It should be called as `log.warn(message)()`. Please create a unit test for this case to verify that it actually works.